fdbserver/DataDistribution.actor.cpp

```diff
@@ -2957,6 +2991,15 @@ ACTOR Future<Void> teamTracker(DDTeamCollection* self, Reference<TCTeamInfo> tea
 
 	try {
 		loop {
+			// teamTracker only works when all teamCollections are valid.
+			// Always check if all teamCollections are valid, and throw error if any teamCollection has been destructed,
+			// because the teamTracker can be triggered after a DDTeamCollection was destroyed and
+			// before the other DDTeamCollection is destroyed.
+			for (int i = 0; i < self->teamCollections.size(); ++i) {
+				if (self->teamCollections[i] == nullptr) {
```

Comment:
do this check when attempting to use the teamCollection later in the function, and throw a different error